### DISCUSSION TITLE
Roteamento no aplicativo

### DISCUSSION THREAD

**Discussion by ribeiroevandro** [diego3g/rsxp-2023#9] [Polls]
Pensando no aplicativo, acho que é pertinente a escolha da lib para lidar com as rotas.

**Comment by FelipeQueiroz** [comment]
Acho que seria mais ideial o expo router devido ao diego já ter optado por utiliza o Expo como stack :)

**Comment by nitoba** [comment]
Expo router pode ser uma boa já que é semelhante ao sistema de rotas do NextJs, porém a documentação dele ainda está um pouco pobre sobre algumas funcionalidades, e conversão sobre organização das pastas e arquivos, tenho percebido o VSCode se perdendo bastante com esse novo diretório app que foi incluído. Mas é uma boa opção para testar

**Comment by RennanD** [comment]
Continuo achando o React Navigation a melhor opção para esse contexto, o Expo Router é uma ferramenta fantástica e que tem muito para entregar, mas acho que ainda é muito "imaturo".

**Reply by nitoba** [reply]
Pesando tudo na balança, o React Navigation é seria a escolha mais segura, concordo com você

**Comment by orodrigogo** [comment]
Pela segurança, eu também curto a ideia do React Navigation. Inclusive, o Expo Router é baseado no Navigation. 🚀

**Comment by diego3g** [comment]
Acho que podemos testar o Expo Router e qualquer coisa migrar! Até porque essa primeira versão do app vai ter, sei lá, 3 rotas?

**Reply by orodrigogo** [reply]
Boa ideia!

**Reply by pironti** [reply]
Acho que esse é o caminho, me lembro quando vi a primeira vez o sistema de rotas do Nextjs (muito parecido com a abordagem do Expo Router) não botei muita fé, mas depois de utilizar o Next em projetos com várias rotas dinâmicas vi o valor que é agregado nesse tipo de roteamento.

Lembrando, que o @diego3g comentou na live do dia 4/4 sobre mantermos a mente aberta e tentarmos criar uma experiencia para novos e experientes dev's com pouca e muita expericencia, logo entendo que a visualização é extremamente simples e só de ver a estrutura de pastas já fica claro o que está acontencendo lá.

E concordo sobre migrar eventualmente, visto que não é uma aplicação que terá MUITAS rotas.

**Reply by leandromatos** [reply]
Só pra pontuar, estava fazendo uns testes com outra aplicação e percebi que a cada troca de rota, todas as rotas são renderizadas novamente.

Tá rolando até uma discussão lá no repositório do expo-router https://github.com/expo/router/issues/320#issuecomment-1509818561.
